### ISSUE TITLE
Validation loss is jumping and indeterministic while restoring the training

### ISSUE THREAD

**Issue by baoruxiao** [matterport/Mask_RCNN#253]
<img width="1127" alt="screen shot 2018-02-13 at 8 05 14 am" src="https://user-images.githubusercontent.com/10040041/36159865-a9247286-1094-11e8-9b3b-08a5deedb55e.png">

The image shows an example of jumping validation loss (especially rpn loss) when the training goes to the next stage. (I was fine-tuning on the provided coco weights)

I use following config to stage the training (notice that I didn't change the learning rates or layers to freeze):
```
        if '1' in args.stage.split(','):
            print("Training network heads")
            model.train(dataset_train, dataset_val,
                        learning_rate=config.LEARNING_RATE/10,
                        epochs=40,
                        layers='heads')

        # Training - Stage 2
        # Finetune layers from ResNet stage 4 and up
        if '2' in args.stage.split(','):
            print("Fine tune Resnet stage 4 and up")
            model.train(dataset_train, dataset_val,
                        learning_rate=config.LEARNING_RATE/10,
                        epochs=60,
                        layers='heads')

        ## Training - Stage 3
        ## Fine tune all layers
        if '3' in args.stage.split(','):
            print("Fine tune all layers")
            model.train(dataset_train, dataset_val,
                        learning_rate=config.LEARNING_RATE/10,
                        epochs=80,
                        layers='heads')
```

Since I saw that you are staging the training for your coco weights, did you observe the same jumping of validation loss when your training transit from one stage to the other?

Please help.

Thanks

**Comment by baoruxiao** [comment]
Same issue as https://github.com/matterport/Mask_RCNN/issues/89

**Comment by lijianaiml** [comment]
@waleedka,@baoruxiao .i have the same question and I didn't change the learning rates or layers to freeze ,the loss curve is the same as above.why does that happen.it borthers me a lot,please help.thanks.

**Comment by lijianaiml** [comment]
@waleedka,@baoruxiao .i have the same question. Since I didn't change the learning rates or layers to freeze ,so the L1 of the weights shouldn't change a lot,but the loss curve is the same as above.
why does that happen.it bother me a lot,please help.thanks.
